### PR TITLE
fix: handle default port in `base`

### LINKED ISSUE #195: `buildURL` doesn't handle `base` with port 80
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

3.21.4

### Plugin version

6.0.1

### Node.js version

14.x

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

11.6

### Description

If `base` contains `:80`, the `buildURL` function throws.

### Steps to Reproduce

```js
const { buildURL } = require('fastify-reply-from/lib/utils')

buildURL('/graphql', 'http://localhost:80')
```

This throws - removing the `:80` works.

(`new URL` strips out the port for `http` URLs (and `443` for `https`))

### Expected Behavior

It should gracefully handle the default port being provided

### PR DESCRIPTION
**Pull request by SimenB**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

Fixes #195

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: lib/utils.js [hunk 1]
```diff
@@ -56,15 +56,16 @@ function stripHttp1ConnectionHeaders (headers) {
 
 // issue ref: https://github.com/fastify/fast-proxy/issues/42
 function buildURL (source, reqBase) {
+  let baseOrigin = reqBase ? new URL(reqBase).href : undefined
```
Comment (SimenB) [inline lib/utils.js:59]: this is using `.href` instead of `.origin` btw, otherwise a bunch of the tests verifying this throws started failing

Comment (mcollina) [reply lib/utils.js:59]: Ok! I'm trusting the tests - do you think there is the possibility of a bug there?

Comment (SimenB) [reply lib/utils.js:59]: Nah, I don't think so. This makes sure the user provides cleaner inputs without "patching" over what is provided which seems reasonable 🙂

**Pull request merged**